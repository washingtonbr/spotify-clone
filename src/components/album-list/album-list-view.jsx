import React from 'react';
import PropTypes from 'prop-types';
import AlbumThumbnail from '../album-thumbnail';
import { Wrapper } from './album-list-styles';

function AlbumListView({ albums }) {
  return (
    <Wrapper>
      {albums.map(({
        id,
        images,
        name,
        artists,
      }) => (
        <AlbumThumbnail
          key={id}
          id={id}
          cover={images[0].url}
          name={name}
          artists={artists.map((artist) => artist.name)}
        />
      ))}
    </Wrapper>
  );
}

AlbumListView.propTypes = {
  albums: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default AlbumListView;
