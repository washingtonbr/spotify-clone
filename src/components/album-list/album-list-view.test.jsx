import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import { albumsMock } from './album-list-view.mock';
import AlbumListView from './album-list-view';
import AlbumThumbnail from '../album-thumbnail';

describe('AlbumListView', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <AlbumListView
            albums={albumsMock}
          />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should display the AlbumThumbnail component 5 times', () => {
    const wrapper = mount(
      <MemoryRouter>
        <AlbumListView
          albums={albumsMock}
        />
      </MemoryRouter>,
    );

    expect(wrapper.find(AlbumThumbnail)).toHaveLength(albumsMock.length);
  });
});
