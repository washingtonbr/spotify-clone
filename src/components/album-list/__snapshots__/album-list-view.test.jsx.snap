// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumListView renders correctly 1`] = `
<div
  className="sc-AxhUy hKptHB"
>
  <article
    className="sc-AxjAm duEsRS"
  >
    <a
      href="/album/4otkd9As6YaxxEkIjXPiZ6"
      onClick={[Function]}
    >
      <img
        alt="Music To Be Murdered By"
        className="sc-AxirZ cTKAUo"
        src="https://i.scdn.co/image/ab67616d0000b2732f44aec83b20e40f3baef73c"
      />
      <h2
        className="sc-AxiKw dfKjlj"
      >
        Music To Be Murdered By
      </h2>
      <h3
        className="sc-AxhCb dxgIcC"
      >
        Eminem
      </h3>
    </a>
  </article>
  <article
    className="sc-AxjAm duEsRS"
  >
    <a
      href="/album/2cWBwpqMsDJC1ZUwz813lo"
      onClick={[Function]}
    >
      <img
        alt="The Eminem Show"
        className="sc-AxirZ cTKAUo"
        src="https://i.scdn.co/image/ab67616d0000b2736ca5c90113b30c3c43ffb8f4"
      />
      <h2
        className="sc-AxiKw dfKjlj"
      >
        The Eminem Show
      </h2>
      <h3
        className="sc-AxhCb dxgIcC"
      >
        Eminem
      </h3>
    </a>
  </article>
  <article
    className="sc-AxjAm duEsRS"
  >
    <a
      href="/album/47BiFcV59TQi2s9SkBo2pb"
      onClick={[Function]}
    >
      <img
        alt="Recovery"
        className="sc-AxirZ cTKAUo"
        src="https://i.scdn.co/image/ab67616d0000b273c08d5fa5c0f1a834acef5100"
      />
      <h2
        className="sc-AxiKw dfKjlj"
      >
        Recovery
      </h2>
      <h3
        className="sc-AxhCb dxgIcC"
      >
        Eminem
      </h3>
    </a>
  </article>
  <article
    className="sc-AxjAm duEsRS"
  >
    <a
      href="/album/3HNnxK7NgLXbDoxRZxNWiR"
      onClick={[Function]}
    >
      <img
        alt="Kamikaze"
        className="sc-AxirZ cTKAUo"
        src="https://i.scdn.co/image/ab67616d0000b273e4073def0c03a91e3fceaf73"
      />
      <h2
        className="sc-AxiKw dfKjlj"
      >
        Kamikaze
      </h2>
      <h3
        className="sc-AxhCb dxgIcC"
      >
        Eminem
      </h3>
    </a>
  </article>
  <article
    className="sc-AxjAm duEsRS"
  >
    <a
      href="/album/6t7956yu5zYf5A829XRiHC"
      onClick={[Function]}
    >
      <img
        alt="The Marshall Mathers LP"
        className="sc-AxirZ cTKAUo"
        src="https://i.scdn.co/image/ab67616d0000b273dbb3dd82da45b7d7f31b1b42"
      />
      <h2
        className="sc-AxiKw dfKjlj"
      >
        The Marshall Mathers LP
      </h2>
      <h3
        className="sc-AxhCb dxgIcC"
      >
        Eminem
      </h3>
    </a>
  </article>
</div>
`;
